src/DiContainer/Autowired.php

```diff
@@ -99,15 +101,51 @@ private function resolveParameters(ContainerInterface $container, array $paramet
             $parameterTypeName = $parameterType->getName();
 
             try {
-                $dependencies[$parameterName] = match (true) {
-                    $container::class === $parameterTypeName,
-                    DiContainerInterface::class === $parameterTypeName => $container,
-                    default => $container->get(
-                        $isBuildIn
+                if ($container::class === $parameterTypeName
+                    || DiContainerInterface::class === $parameterTypeName) {
+                    $dependencies[$parameterName] = $container;
+
+                    continue;
+                }
+
+                if ($attribute = ($parameter->getAttributes(Inject::class)[0] ?? null)) {
```

Comment:
Think about make code short (separate to different methods) 🤔
Code for resolving by PHP attribute.